common.mk

```
@@ -33,6 +33,16 @@ ifneq ($(shell python -c "import wheel as w; \
 $(error Looks like the `wheel` package is outdated or missing. See README for instructions on how to fix this.)
 endif
 
+# This check should not occur within CI environments, where AWS Credentials might not be supplied
+ifneq ($(shell python -c "import os, sys; \
+                          import boto3 as b; \
+                          print(True) if os.getenv('CI') else \
+                          print(b.client('sts').get_caller_identity()['Account'] == \
+                          os.environ['AZUL_AWS_ACCOUNT_ID'])"),True)
```

```suggestion
ifneq ($(shell python -c "import os, sys; \
                          import boto3 as b; \
                          print(os.getenv('CI') or \
                          b.client('sts').get_caller_identity()['Account'] == os.environ['AZUL_AWS_ACCOUNT_ID'])"),True)
```

and don't mix os.getenv with os.environ

---

@hannes-ucsc i'll switch to getenv for both.

---

I should have been more clear. There is precedence in common.mk and the rest of Azul for `os.environ` and you should use that. The reason we prefer `os.environ` is that we can more clearly express whether we do or do not expect a variable to be present(`os.environ['…']` vs `os.environ.get('…')`).

---

in this case i dont expect an operators machine to have the env var `CI` set.
In that case if we were to use `os.environ['CI']` there would be KeyError raised when an operator would run any Make command. 

I could perform additional checks within the env files that could set CI to False or True,and move to explicitly check if CI is set to True. However there would be additional complexity added for both Travis and Gitlab. 

Unless there is an easier way. 🤔

---

My post on Slack should resolve this. Let me know if it does not.

---

@hannes-ucsc  There was a new issue with using the `or` statement within the print, Travis and Gitlab both set the CI value to `true` while our comparison is for a boolean `True`; it ends up always = failin due to `true != True`. 
Was considering using https://docs.python.org/3/distutils/apiref.html#distutils.util.strtobool to get around this.

---

`strtobool` fails to handle None case, considering; 
```
('true' if b.client('sts').get_caller_identity()['Account'] == os.environ['AZUL_AWS_ACCOUNT_ID']))\
```
and setting the comparison to `true` within the make statement. 

```
ifneq ($(shell python -c "import os, sys; \
                          import boto3 as b; \
                          print(os.environ.get('CI') or \
                          ('true' if b.client('sts').get_caller_identity()['Account'] == os.environ['AZUL_AWS_ACCOUNT_ID']))\
                          "),true)
```

---

I'll reply on the most recent changes.